---
title: "BloodGen3Module: Modular Repertoire Analysis and Visualization"
author: "Darawan Rinchai"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{introduction}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(echo = T,
                      results = "hide")

```

***
The **BloodGen3Module** package provides functions for R user performing module repertoire analyses and generating fingerprint representations.

Steps involved in module repertoire analysis and visualization include: 

1.	The annotation of the gene expression data matrix with module membership information. 
2.	Running of statistical tests to determine for each module the proportion of its constitutive genes which are differentially expressed.
3.	The results are expressed “at the module level” as percent of genes increased or decreased. 
4.	Visualization in a fingerprint grid format for group comparison and fingerprint heatmap for individual level analysis.

## Installation
***
The recommended installation for the R package is to use the install_github function from the devtools package.

```{r Package installation}
#install.packages("devtools")
#devtools::install_github("Drinchai/BloodGen3Module")

```

## Usage
  ***
##Load library
These are list of required R packages that must be installed before the analysis.

```{r setup, warning=FALSE,message=FALSE}
library(BloodGen3Module)

```

##Arguments

data.matrix      Normalized expression data (not Log2 transformed as it will be automatic transformed when running this function)
sample_info      A table of sample annotation 
FC               Foldchange cut off to consider th eabundance of a given transcript to be increased or decreased compared to a reference group (Ref_group)
DIFF             Difference cut off to consider th eabundance of a given transcript to be increased or decreased compared to a reference group (Ref_group)
pval             p-value cut off or False discovery rate when FDR = FALSE
FDR              False discovery rate cut off
Group_column     Name of the columns for the groups used for the analysis
Ref_group        Reference group or samples that considered as control 
Group_df         Output table generated after running the 'Groupcomparison' function 
Individual_df    Output table generated after running the 'Individualcomparison' function
cutoff           Sets the percentage cut off used for fingerprint visualization, range of acceptable values from 0 to 100
rowSplit         Splits row of heatmap by each aggregate 
show_ref_group	 Plot reference group in the heatmap
Aggregate        Selects specific module aggregates for heatmap fingerprint plot
filename         Give a name for saving file
height           Sets height dimension for the heatmap plot
width            Sets width dimension for the heatmap plot




## Input
  ***
To perform the modular repertoire analysis, the R package simply requires sample annotation table and the matrix format of raw transcriptome data as inputs. Expression data should be normalized before starting the analysis. In this example, the quantile normalization was applied using function "normalize.quantiles from library(preprocessCore). Sample annotation file must re-named to sample.info after loading into R.

Examples data could be downloaded here; https://github.com/Drinchai/BloodGen3Module/tree/master/data.

```{r raw data and annotaion preparation}
#Load expression data
#Expression data
data.matrix = data_exp

#Sample annotation file
sample.info = sample_ann

```

## Group comparison analysis 
***
Groupcomparison function will perform group comparison analysis and the results are expressed “at the module level” as percent of genes increased or decreased.Annotation of the gene expression data matrix with module membership information. Then differential expression and calculation of percentage response. 
-Expression matrix and sample annotation file are required for performing this analysis. 
-The sample annotation file must load as specific name = "sample.info". 
-A specific column of conditions for analysis must be indicated

## Using t-test statistical analysis
```{r group comparison analysis using t-test statistical analysis,warning=FALSE}
Group_df <- Groupcomparison(data.matrix,
                            FC = 1.5,
                            pval = 0.1 ,
                            FDR = TRUE,
                            Group_column = "Group_test",
                            Ref_group = "Control")
```

## Using limma statistical analysis
```{r group comparison analysis using limma statistical analysis,warning=FALSE}
Group_limma <- Groupcomparison.limma(data.matrix,
                            FC = 1.5,
                            pval = 0.1 ,
                            FDR = TRUE,
                            Group_column = "Group_test",
                            Ref_group = "Control")
```

## Fingerprint grid visualization 
***
gridplot function will perform grid plot in a pdf file. Specific working directory for the analysis need to be specific for saving the file. The result of the plot should be return in the same working directory.

Cut off for visualization is set default at percentage of response > 15, it can be changed from 0-100%. 

## After running "Groupcomparison" function
```{r grid visulization after running "Groupcomparison" function}

gridplot(Group_df, 
         cutoff = 15, 
         Ref_group = "Control",
         filename="Group_comparison_cutoff15")

```


## After running "Groupcomparison.limma" function
```{r grid visulization after running "Groupcomparison.limma" function}

gridplot.limma(Group_limma, 
               cutoff = 15, 
               Ref_group = "Control",
               filename="Limma_group_comparison")

```

## Individual single sample analysis 
***
Individualcomparison function will perform single sample analysis as comparison to control and the results are expressed “at the module level” as percent of genes increased or decreased. Annotation of the gene expression data matrix with module membership information. Then differential expression and calculation of percentage response. 

-Expression matrix and sample annotation file are required for performing this analysis. 
-The sample annotation file must load as specific name = "sample.info"
-A specific column of conditions for analysis must be indicated
-Default is setting FC =1.5 and DIFF =10 


```{r individual single sample analysis, warning=FALSE}
Individual_df = Individualcomparison(data.matrix,
                                     FC = 1.5,
                                     DIFF = 10,
                                     Group_column = "Group_test",
                                     Ref_group = "Control")
```

## Individual fingerprint visualization 
***
fingerprintplot function will perform fingerprint heatmap plot in a pdf file. Specific working directory for the analysis need to be specific for saving the file. The result of the plot should be return in the same working directory.

Cut off for visualization is set at percentage of response > 15, this cut off can be changed from 0-100%. 
 

```{r fingerprint visualization, warning=FALSE}

fingerprintplot(Individual_df,
                cutoff = 15,
                rowSplit= TRUE ,
                Group_column= "Group_test",
                Ref_group =  "Control",
                Aggregate = c("A28"),
                height = NULL,
                width = NULL)

```

##Notes
***
•	It is important to note that sample annotation file must re-named to sample.info after loading into R.

##Publication
***
A manuscript is currently under consideration for publication, to cite currently please refer to the bioRxiv preprint:
https://www.biorxiv.org/content/10.1101/2020.07.16.205963v1
