---
title: "BloodGen3Module: Modular Repertoire Analysis and Visualization"
author: "Darawan Rinchai"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{BloodGen3Module: Modular Repertoire Analysis and Visualization}
  %\VignetteEngine{knitr::rmarkdown}
  \usepackage[utf8]{inputenc}
---

```{r, echo = FALSE, eval = FALSE}
library(knitr)
knitr::opts_chunk$set(
    error = FALSE,
    tidy  = FALSE,
    message = FALSE,
    warning = FALSE,
    fig.align = "center"
)
```

***
The **BloodGen3Module** package provides functions for R user performing module repertoire analyses and generating fingerprint representations.

Steps involved in module repertoire analysis and visualization include:

1. Annotating the gene expression data matrix with module membership information.\
2. Running statistical tests to determine for each module the proportion of constitutive genes which are differentially expressed.\
3. Expressing results “at the module level” as percent of genes increased or decreased.\
4. Visualizing results from group comparison as a fingerprint grid and results from individual sample comparisons as a fingerprint heatmap.\

## Installation
***
It is recommended to use the install_github function from the devtools package in order to install the R package.

```{r Package installation, echo= TRUE}
#install.packages("devtools")
#devtools::install_github("Drinchai/BloodGen3Module")

```

## Usage
  ***
##Load library

```{r setup, warning=FALSE,message=FALSE, echo= TRUE}
library(BloodGen3Module)

```

## Input
  ***
To perform the modular repertoire analysis, the R package simply requires a sample annotation table and a normalized expression data matrix For illustrative purposes sample input files can be downloaded here; https://github.com/Drinchai/DC_Gen3_Module_analysis/tree/master/R%20data.

```{r raw data and annotaion preparation, warning=FALSE,message=FALSE,echo= TRUE}
#Load expression data
#Example expression data for package testting

Test_sample =  matrix(data = rexp(1000, rate = 0.01), nrow = 14168, ncol = 20)
control_sample = matrix(data = rexp(1000, rate = 0.1), nrow = 14168, ncol = 10)

data.matrix = data.frame(cbind(Test_sample,control_sample))
data.matrix$Symbol = Module_listGen3$Gene
data.matrix = aggregate(data.matrix,FUN = mean,by=list(data.matrix$Symbol))
rownames(data.matrix) = data.matrix$Group.1
data.matrix$Group.1 = NULL
data.matrix$Symbol = NULL
colnames(data.matrix) = c(paste0(rep("SampleID",30),1:30))

##example sample information
sample_ann = data.frame(SampleID=(colnames(data.matrix)),Group_test = c(rep("Test",20),rep("Control",10)),stringsAsFactors = F)
rownames(sample_ann) = sample_ann$SampleID
rownames(sample_ann) == colnames(data.matrix)
head(sample_ann)

```

## Group comparison analysis 
***
The Groupcomparison function will perform group comparison analyses and the results are expressed “at the module level” as percent of genes increased or decreased.

- Expression matrix and sample annotation files are required to perform this analysis.
- The names of the columns for the conditions used in the analysis must be specified.

## Using t-test statistical analysis
```{r group comparison analysis using t-test statistical analysis, warning=FALSE, echo= TRUE}
Group_df <- Groupcomparison(data.matrix,
                            sample_info = sample_ann,
                            FC = 1.5,
                            pval = 0.1 ,
                            FDR = TRUE,
                            Group_column = "Group_test",
                            Ref_group = "Control")
```

## Using limma statistical analysis
```{r group comparison analysis using limma statistical analysis,warning=FALSE}
Group_limma <- Groupcomparisonlimma(data.matrix,
                                    sample_info = sample_ann,
                                    FC = 1.5,
                                    pval = 0.1 ,
                                    FDR = TRUE,
                                    Group_column = "Group_test",
                                    Test_group = "Sepsis",
                                    Ref_group = "Control")
```



## Fingerprint grid visualization 
***
The gridplot function will generate a grid plot as a pdf file. Specific working directory for the analysis need to be specified for saving the file. The result of the plot should be return in the same working directory.

The default cut off for visualization is set at 15%, it can changed to any value between 0-100%.

## After running "Groupcomparison" function
```{r grid visulization after running "Groupcomparison" function}
gridplot(Group_df, 
         cutoff = 15, 
         Ref_group = "Control",
         filename="Group_comparison_cutoff15")
```


## After running "Groupcomparison.limma" function
```{r grid visulization after running "Groupcomparison.limma" function}
gridplotlimma(Group_limma, 
               cutoff = 15, 
               Ref_group = "Control",
               filename="Limma_group_comparison")
```

## Individual single sample analysis 
***
The Individualcomparison function will perform individual sample comparison analysis in reference to a control sample or group of samples, with the results are expressed “at the module level” as percent of genes increased or decreased.

- Expression matrix and sample annotation file are required in order to perform this analysis.
- The names of the columns for the conditions used in the analysis must be specified
- The default cutoff is set at FC =1.5 and DIFF =10


```{r individual single sample analysis, warning=FALSE, echo= TRUE}
Individual_df = Individualcomparison(data.matrix,
                                     sample_info = sample_ann,
                                     FC = 1.5,
                                     DIFF = 10,
                                     Group_column = "Group_test",
                                     Ref_group = "Control")
```

##Notes
***
•	It is important to note that:\
- Expression matrix must be none Log2 transformed as it will be automatic transformed when running this function.\
- rownames of sample information (sample_info) must be the same names as in colnames of data.matrix

##Publication
***
A manuscript is currently under consideration for publication, to cite currently please refer to the bioRxiv preprint:\
https://www.biorxiv.org/content/10.1101/2020.07.16.205963v1
